src/components/Context/injectWeb.js

```diff
@@ -12,14 +12,20 @@ governing permissions and limitations under the License.
 
 import { deepAssign } from "../../utils";
 
+// URLs reported by browser APIs may not be URI-compliant because browsers are more lenient
+// than the URI RFC (https://tools.ietf.org/html/rfc3986). The XDM fields we're populating,
+// however, should conform to the URI spec. For this reason, we need to encode the URL.
+// We decode the URL first so it doesn't end up double encoded if it was previously encoded.
+const encode = url => encodeURI(decodeURI(url));
```

Comment:
By doing this we'll avoid double encoding, but we lose a way to provide a representation of the original url we obtained. This is probably less important than avoiding multiple encoding but something we need to be aware of.

Reply:
Thanks @liljenback. I acknowledge this is the case. Do you have any ideas on how to more appropriately handle this situation?

Reply:
I think it'll be fine to do it this way. Avoiding multiple encodings will prevent security related problems. Keeping it simple is most important.